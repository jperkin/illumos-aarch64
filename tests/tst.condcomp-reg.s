/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.5.5 Conditional compare (register)
 */
.macro condcomp_reg op
	/* 32-bit */
	\op	w0,  w0,  #0,  eq
	\op	w0,  wzr, #1,  ne
	\op	wzr, w0,  #2,  cs
	\op	wzr, wzr, #3,  cc
	\op	w0,  w0,  #4,  mi
	\op	w0,  wzr, #5,  pl
	\op	wzr, w0,  #6,  vs
	\op	wzr, wzr, #7,  vc
	\op	w0,  w0,  #8,  hi
	\op	w0,  wzr, #9,  ls
	\op	wzr, w0,  #10, ge
	\op	wzr, wzr, #11, lt
	\op	w0,  w0,  #12, gt
	\op	w0,  wzr, #13, le
	\op	wzr, w0,  #14, al
	\op	wzr, wzr, #15, nv
	/* 64-bit */
	\op	x0,  x0,  #0,  eq
	\op	x0,  xzr, #1,  ne
	\op	xzr, x0,  #2,  cs
	\op	xzr, xzr, #3,  cc
	\op	x0,  x0,  #4,  mi
	\op	x0,  xzr, #5,  pl
	\op	xzr, x0,  #6,  vs
	\op	xzr, xzr, #7,  vc
	\op	x0,  x0,  #8,  hi
	\op	x0,  xzr, #9,  ls
	\op	xzr, x0,  #10, ge
	\op	xzr, xzr, #11, lt
	\op	x0,  x0,  #12, gt
	\op	x0,  xzr, #13, le
	\op	xzr, x0,  #14, al
	\op	xzr, xzr, #15, nv
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.33 CCMN (register)
	 * C5.6.35 CCMP (register)
	 */
	condcomp_reg ccmn
	condcomp_reg ccmp
.size libdis_test, [.-libdis_test]
