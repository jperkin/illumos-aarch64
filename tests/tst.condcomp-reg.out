    libdis_test:       00 00 40 3a  ccmn w0, w0, #0x0, eq
    libdis_test+0x4:   01 10 5f 3a  ccmn w0, wzr, #0x1, ne
    libdis_test+0x8:   e2 23 40 3a  ccmn wzr, w0, #0x2, cs
    libdis_test+0xc:   e3 33 5f 3a  ccmn wzr, wzr, #0x3, cc
    libdis_test+0x10:  04 40 40 3a  ccmn w0, w0, #0x4, mi
    libdis_test+0x14:  05 50 5f 3a  ccmn w0, wzr, #0x5, pl
    libdis_test+0x18:  e6 63 40 3a  ccmn wzr, w0, #0x6, vs
    libdis_test+0x1c:  e7 73 5f 3a  ccmn wzr, wzr, #0x7, vc
    libdis_test+0x20:  08 80 40 3a  ccmn w0, w0, #0x8, hi
    libdis_test+0x24:  09 90 5f 3a  ccmn w0, wzr, #0x9, ls
    libdis_test+0x28:  ea a3 40 3a  ccmn wzr, w0, #0xa, ge
    libdis_test+0x2c:  eb b3 5f 3a  ccmn wzr, wzr, #0xb, lt
    libdis_test+0x30:  0c c0 40 3a  ccmn w0, w0, #0xc, gt
    libdis_test+0x34:  0d d0 5f 3a  ccmn w0, wzr, #0xd, le
    libdis_test+0x38:  ee e3 40 3a  ccmn wzr, w0, #0xe, al
    libdis_test+0x3c:  ef f3 5f 3a  ccmn wzr, wzr, #0xf, nv
    libdis_test+0x40:  00 00 40 ba  ccmn x0, x0, #0x0, eq
    libdis_test+0x44:  01 10 5f ba  ccmn x0, xzr, #0x1, ne
    libdis_test+0x48:  e2 23 40 ba  ccmn xzr, x0, #0x2, cs
    libdis_test+0x4c:  e3 33 5f ba  ccmn xzr, xzr, #0x3, cc
    libdis_test+0x50:  04 40 40 ba  ccmn x0, x0, #0x4, mi
    libdis_test+0x54:  05 50 5f ba  ccmn x0, xzr, #0x5, pl
    libdis_test+0x58:  e6 63 40 ba  ccmn xzr, x0, #0x6, vs
    libdis_test+0x5c:  e7 73 5f ba  ccmn xzr, xzr, #0x7, vc
    libdis_test+0x60:  08 80 40 ba  ccmn x0, x0, #0x8, hi
    libdis_test+0x64:  09 90 5f ba  ccmn x0, xzr, #0x9, ls
    libdis_test+0x68:  ea a3 40 ba  ccmn xzr, x0, #0xa, ge
    libdis_test+0x6c:  eb b3 5f ba  ccmn xzr, xzr, #0xb, lt
    libdis_test+0x70:  0c c0 40 ba  ccmn x0, x0, #0xc, gt
    libdis_test+0x74:  0d d0 5f ba  ccmn x0, xzr, #0xd, le
    libdis_test+0x78:  ee e3 40 ba  ccmn xzr, x0, #0xe, al
    libdis_test+0x7c:  ef f3 5f ba  ccmn xzr, xzr, #0xf, nv
    libdis_test+0x80:  00 00 40 7a  ccmp w0, w0, #0x0, eq
    libdis_test+0x84:  01 10 5f 7a  ccmp w0, wzr, #0x1, ne
    libdis_test+0x88:  e2 23 40 7a  ccmp wzr, w0, #0x2, cs
    libdis_test+0x8c:  e3 33 5f 7a  ccmp wzr, wzr, #0x3, cc
    libdis_test+0x90:  04 40 40 7a  ccmp w0, w0, #0x4, mi
    libdis_test+0x94:  05 50 5f 7a  ccmp w0, wzr, #0x5, pl
    libdis_test+0x98:  e6 63 40 7a  ccmp wzr, w0, #0x6, vs
    libdis_test+0x9c:  e7 73 5f 7a  ccmp wzr, wzr, #0x7, vc
    libdis_test+0xa0:  08 80 40 7a  ccmp w0, w0, #0x8, hi
    libdis_test+0xa4:  09 90 5f 7a  ccmp w0, wzr, #0x9, ls
    libdis_test+0xa8:  ea a3 40 7a  ccmp wzr, w0, #0xa, ge
    libdis_test+0xac:  eb b3 5f 7a  ccmp wzr, wzr, #0xb, lt
    libdis_test+0xb0:  0c c0 40 7a  ccmp w0, w0, #0xc, gt
    libdis_test+0xb4:  0d d0 5f 7a  ccmp w0, wzr, #0xd, le
    libdis_test+0xb8:  ee e3 40 7a  ccmp wzr, w0, #0xe, al
    libdis_test+0xbc:  ef f3 5f 7a  ccmp wzr, wzr, #0xf, nv
    libdis_test+0xc0:  00 00 40 fa  ccmp x0, x0, #0x0, eq
    libdis_test+0xc4:  01 10 5f fa  ccmp x0, xzr, #0x1, ne
    libdis_test+0xc8:  e2 23 40 fa  ccmp xzr, x0, #0x2, cs
    libdis_test+0xcc:  e3 33 5f fa  ccmp xzr, xzr, #0x3, cc
    libdis_test+0xd0:  04 40 40 fa  ccmp x0, x0, #0x4, mi
    libdis_test+0xd4:  05 50 5f fa  ccmp x0, xzr, #0x5, pl
    libdis_test+0xd8:  e6 63 40 fa  ccmp xzr, x0, #0x6, vs
    libdis_test+0xdc:  e7 73 5f fa  ccmp xzr, xzr, #0x7, vc
    libdis_test+0xe0:  08 80 40 fa  ccmp x0, x0, #0x8, hi
    libdis_test+0xe4:  09 90 5f fa  ccmp x0, xzr, #0x9, ls
    libdis_test+0xe8:  ea a3 40 fa  ccmp xzr, x0, #0xa, ge
    libdis_test+0xec:  eb b3 5f fa  ccmp xzr, xzr, #0xb, lt
    libdis_test+0xf0:  0c c0 40 fa  ccmp x0, x0, #0xc, gt
    libdis_test+0xf4:  0d d0 5f fa  ccmp x0, xzr, #0xd, le
    libdis_test+0xf8:  ee e3 40 fa  ccmp xzr, x0, #0xe, al
    libdis_test+0xfc:  ef f3 5f fa  ccmp xzr, xzr, #0xf, nv
